test/regress/run_regression.py

```diff
@@ -223,6 +251,8 @@ def run_regression(unsat_cores, proofs, dump, use_skip_return_code, wrapper,
             sys.exit('Cannot determine status of "{}"'.format(benchmark_path))
     if expected_exit_status is None:
         expected_exit_status = 0
+    if logic_regex:
+        logic = re.findall(logic_regex, benchmark_content)
```

Comment:
This should be:

```suggestion
        logic_match = re.search(logic_regex, benchmark_content)
        if logic_match:
            logic = logic_match[1]
```

I think right now the `logic_supported_with_proofs` always fails because we are giving it a list of matches instead of a string.